Index: asmc.c
===================================================================
--- asmc.c	(revision 303634)
+++ asmc.c	(working copy)
@@ -206,6 +206,12 @@
 	  ASMC_MBP8_TEMPS, ASMC_MBP8_TEMPNAMES, ASMC_MBP8_TEMPDESCS
 	},
 
+        {
+         "MacBookPro9,2", "Apple SMC MacBook Pro (mid 2012)",
+          ASMC_SMS_FUNCS, ASMC_FAN_FUNCS, ASMC_LIGHT_FUNCS,
+          ASMC_MBP9_TEMPS, ASMC_MBP9_TEMPNAMES, ASMC_MBP9_TEMPDESCS
+        },
+
 	{ 
 	  "MacBookPro11,3", "Apple SMC MacBook Pro Retina Core i7 (2013/2014)",
 	  ASMC_SMS_FUNCS, ASMC_FAN_FUNCS, ASMC_LIGHT_FUNCS,
Index: asmcvar.h
===================================================================
--- asmcvar.h	(revision 303634)
+++ asmcvar.h	(working copy)
@@ -197,6 +197,25 @@
 				  "TMBS", "TP0P", "TPCD", "TW0P", "Th1H", \
 				  "Th2H", "Tm0P", "Ts0P", "Ts0S" } 
 
+#define ASMC_MBP9_TEMPS         { "Ts0P", "Ts0S", "TA0P", "TB1T", "TB2T", \
+                                  "TB0T", "TC1C", "TC2C", "TC0E", "TC0F", \
+                                  "TC0J", "TC0P", "TCFC", "TCGC", "TCSA", \
+                                  "TCTD", "TCXC", "TG1D", "TM0P", "TM0S", \
+                                  "TPCD", NULL }
+
+#define ASMC_MBP9_TEMPNAMES     { "Ts0P", "Ts0S", "TA0P", "TB1T", "TB2T", \
+                                  "TB0T", "TC1C", "TC2C", "TC0E", "TC0F", \
+                                  "TC0J", "TC0P", "TCFC", "TCGC", "TCSA", \
+                                  "TCTD", "TCXC", "TG1D", "TM0P", "TM0S", \
+                                  "TPCD" }
+
+#define ASMC_MBP9_TEMPDESCS     { "Palm Rest", "Memory Proximity", "Airflow 1", "Battery 1", "Battery 2", \
+                                  "Battery TS_MAX", "CPU Core 1", "CPU Core 2", "CPU1", "CPU1", \
+                                  "TC0J", "CPU 1 Proximity", "TCFC", "PECI GPU", "PECI SA", \
+                                  "TCTD", "PECI CPU", "GPU Die", "Memory Bank A1", "Memory Module A1", \
+                                  "PCH Die" }
+
+
 #define ASMC_MBP11_TEMPS	{ "TB0T", "TB1T", "TB2T", "TBXT", "TC0E", \
 				  "TC0F", "TC0P", "TC1C", "TC2C", "TC3C", \
 				  "TC4C", "TCFC", "TCGC", "TCSA", "TCTD", \
